srml/balances/src/lib.rs

```diff
@@ -370,26 +374,26 @@ impl<T: Trait> Module<T> {
 
 	/// Returns `Ok` iff the account is able to make a withdrawal of the given amount
 	/// for the given reason.
-	/// 
+	///
 	/// `Err(...)` with the reason why not otherwise.
 	pub fn ensure_account_can_withdraw(
 		who: &T::AccountId,
-		_amount: T::Balance,
-		reason: WithdrawReason,
+		reasons: impl Into<WithdrawReasons>,
```

Comment:
I never encounter this syntax before. Can I know where I can read more about this syntax?

Reply:
in parameter as far as I know it is same as if I have a generic in function `T: Into<WithdrawReasons>` and then `reasons: T` I don't have that much material maybe try to google `impl trait`

Reply:
Incorrect - there should be just one `WithdrawReason` for any single withdraw,

Reply:
Let's say I want to create a runtime module that allow user to reserve and transfer in a single runtime call. How can do it? Call `ensure_account_can_withdraw` twice with different reasons before all the withdraws will not give the correct result.

Reply:
actually calling 
```rust
ensure_account_can_withdraw(who, Reserve | Transfer, new_free_balance)
```
is equivalent to
```rust
ensure_account_can_withdraw(who, Transfer, new_free_balance)
ensure_account_can_withdraw(who, Reserve, new_free_balance)
```